### DISCUSSION TITLE
Generalized usage of idealOrdering for coordinates

### DISCUSSION THREAD

**Discussion by aegershman** [openrewrite/rewrite#257] [Ideas] [answered]
Yammering this out, but: I've been thinking about the usage of `Comparator` when inserting a new annotation on `addAnnotation` coordinates:

https://github.com/openrewrite/rewrite/blob/d21da2e8b167e18c2339f05a2457572c09bf8f6a/rewrite-java/src/main/java/org/openrewrite/java/tree/Coordinates.java#L193-L195

I can think of situations where it'd be valuable to add a new field or method decl to a class; could specifying some sort of comparator or ideal insertion ordering help in the general case? As in, if I wanted to insert a new `J.MethodDeclaration` back into a `J.ClassDeclaration` body, the _easiest_ way to insert it is using the coordinates of `classDeclaration.getBody().getCoordinates().lastStatement()`, or finding a way to do `before()` on any existing `Statements` in the body (if any exist). 

What would, if any, a generalized usage of idealordering look like? If I was inserting a new MethodDeclaration, then nn a ClassDeclaration, could I specify `addStatement()` as the coordinates, and be able to say "insert this in where Method Declarations _should_ go as first, then alphabetically within method declarations" (or have that be specified by some sort of predefined general style)?

Just putting the thought out there 👍

**Comment by jkschneider** [comment, marked as answer]
Looking to add a method to `ListUtils` to generalize this:

![image](https://user-images.githubusercontent.com/1697736/107568427-cf7b4f80-6b9b-11eb-819e-6d8a3a71cb44.png)
